command/ca/renew.go

```diff
@@ -558,6 +576,44 @@ func (r *renewer) Daemon(outFile string, next, expiresIn, renewPeriod time.Durat
 	}
 }
 
+// RenewAfterExpiry creates an authorization token with the given certificate
+// and attempts to renew the expired certificate.
+func (r *renewer) RenewAfterExpiry(cert tls.Certificate) (*api.SignResponse, error) {
+	var issuer string
+	if ext, ok := provisioner.GetProvisionerExtension(cert.Leaf); ok {
+		issuer = ext.Name
+	}
+	claims, err := token.NewClaims(
```

Comment:
Why aren't we reusing the code from `generateNewToken` here?

Reply:
Because `generateNewToken` reads the certificate and key from a file pointed by flags. In the `step ca renew` method the certificate and keys are also backed by files, but for example, in a renewal daemon everything is in memory and we write new certificates to a specific file, but after renewal, the user can run a script that might move the file to a different location, so there's no guarantee that the files are going to be there.